nixos/roles/matomo.nix

```diff
@@ -0,0 +1,101 @@
+{ config, lib, pkgs, ... }:
+
+with builtins;
+
+let
+  inherit (config) fclib;
+  cfg = config.flyingcircus.roles.matomo;
+  serviceCfg = config.services.matomo;
+
+  inherit (serviceCfg.tools) matomoCheckPermissions matomoConsole;
+
+  currentMemory = fclib.currentMemory 1024;
+
+  phpFpmMemoryLimit = fclib.min [ (currentMemory * 25 / 100) 1024 ];
+in
+{
+  options = with lib; {
+    flyingcircus.roles.matomo = {
+      enable = mkEnableOption "Matomo Web Analytics";
+
+      supportsContainers = fclib.mkEnableContainerSupport;
+
+      hostname = mkOption {
+        type = types.str;
+        default = fclib.fqdn { vlan = "fe"; };
+        description = ''
+          Public FQDN for the Matomo Web UI.
+          A Letsencrypt certificate is generated for it.
+          Defaults to the FE FQDN.
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    flyingcircus.passwordlessSudoRules = [
+      {
+        commands = [
+          "${serviceCfg.tools.matomoConsole}/bin/matomo-console"
+        ];
+        users = [ "sensuclient" ];
+        groups = ["service" ];
+        runAs = "matomo";
+      }
+      {
+        commands = [
+          "${serviceCfg.tools.matomoCheckPermissions}/bin/matomo-check-permissions"
+          "${pkgs.coreutils}/bin/stat /var/lib/matomo/share/config/config.ini.php"
+        ];
+        users = [ "sensuclient" ];
+        groups = ["service" ];
+      }
+    ];
+
+    flyingcircus.services.sensu-client.checks = {
+
+      matomo-config = {
+        notification = "Config file cannot be read.";
+        command = ''
+          if ! sudo -u matomo stat /var/lib/matomo/share/config/config.ini.php; then
+            echo "config.ini.php not found, is Matomo installed?"
+            exit 1
+          fi
+        '';
+      };
+      matomo-permissions = {
+        notification = "Matomo permissions are wrong";
+        command = ''
+          sudo ${matomoCheckPermissions}/bin/matomo-check-permissions
+        '';
+      };
+
+      matomo-unexpected-files = {
+        notification = "Cannot get Matomo version";
```

Comment:
This notification is wrong as well, had been mixed up with the one below (before it got fixed).